packages/fuelet_secure_layer/lib/src/features/password/password_manager.dart

```diff
@@ -67,46 +71,184 @@ class PasswordManager {
+    await _secureStorage.write(
+      key: _passwordKey,
+      value: secretToStore,
+      aOptions: const AndroidOptions(
+        encryptedSharedPreferences: false,
+      ),
+    );
+    await _secureStorage.delete(
+      key: _legacyPasscodeKey,
+      aOptions: const AndroidOptions(
+        encryptedSharedPreferences: false,
+      ),
+    );
+    await _sessionStoragePasswordManager.storeSessionStoragePassword(password);
+  }
 
+  Future<void> setSessionStoragePassword(String password) async {
     await _sessionStoragePasswordManager.storeSessionStoragePassword(password);
   }
 
-  Future<AuthorizationResponse> authorize(String input) async {
-    final passwordString = await _secureStorage.read(key: _passwordKey);
-    // If passwordString is null, it means that the user might be using the legacy passcode policy
-    if (passwordString == null) {
-      final passcode = await _secureStorage.read(key: _legacyPasscodeKey);
-      if (passcode == null) {
-        return AuthorizationResponse.noPassword;
+  Future<void> storePasswordForBiometry(String password) async {
+    validatePassword(password);
+
+    if (Platform.isIOS) {
+      final enclave = SecureEnclave();
+
+      final keyResult = await _tpmService.generateKeyPair(_secureEnclaveKey);
+      if (keyResult.isLeft()) {
+        throw PasswordManagementException('Failed to generate Secure Enclave key');
       }
-      if (passcode == input) {
-        return AuthorizationResponse.correctLegacyPasscode;
+
+      final result = await enclave.encrypt(
+        message: password,
+        tag: _secureEnclaveKey,
+      );
+
+      if (result.error != null || result.value == null) {
+        throw PasswordManagementException('Failed to encrypt with Secure Enclave');
       }
 
+      final encryptedBytes = result.value;
+      final base64 = base64Encode(encryptedBytes!);
+
+      await _secureStorage.write(
+        key: _biometricPasswordKey,
+        value: base64,
+        iOptions: const IOSOptions(
+          accessibility: KeychainAccessibility.unlocked,
+        ),
+        aOptions: const AndroidOptions(
+          encryptedSharedPreferences: true,
+        ),
+      );
+    } else if (Platform.isAndroid) {
+      final base64 = base64Encode(utf8.encode(password));
+      await _secureStorage.write(
+        key: _biometricPasswordKey,
+        value: base64,
+        aOptions: const AndroidOptions(
+          encryptedSharedPreferences: false,
+        ),
+      );
+      final secretToStore = await _createPasswordSecret(password);
+      await _secureStorage.write(
+        key: _passwordKey,
+        value: secretToStore,
+        aOptions: const AndroidOptions(
+          encryptedSharedPreferences: false,
+        ),
+      );
+      await _secureStorage.delete(
+        key: _legacyPasscodeKey,
+        aOptions: const AndroidOptions(
+          encryptedSharedPreferences: false,
+        ),
+      );
+      await setSessionStoragePassword(password);
```

Comment:
We probably don't need to handle all of this when storing keys for biometry. And especially if `Platform.isAndroid` only